CondFormats/CTPPSReadoutObjects/plugins/CTPPSPixelDAQMappingESSourceXML.cc

```diff
@@ -0,0 +1,585 @@
+  subSystemName(conf.getUntrackedParameter<string>("subSystem")),
+  currentBlock(0),
+  currentBlockValid(false)
+{
+  for (const auto it : conf.getParameter<vector<ParameterSet>>("configuration"))
+    {
+      ConfigBlock b;
+      b.validityRange = it.getParameter<EventRange>("validityRange");
+      b.mappingFileNames = it.getParameter< vector<string> >("mappingFileNames");
+      b.maskFileNames = it.getParameter< vector<string> >("maskFileNames");
+      configuration.push_back(b);
+    }
+
+  setWhatProduced(this, &CTPPSPixelDAQMappingESSourceXML::produceCTPPSPixelAnalysisMask, es::Label(subSystemName));
+  findingRecord<CTPPSPixelAnalysisMaskRcd>();
+
+  setWhatProduced(this, &CTPPSPixelDAQMappingESSourceXML::produceCTPPSPixelDAQMapping, es::Label(subSystemName));
+  findingRecord<CTPPSPixelDAQMappingRcd>();
+
+
+  LogVerbatim("CTPPSPixelDAQMappingESSourceXML") << " Inside  CTPPSPixelDAQMappingESSourceXML";
+
+
+}
+
+void CTPPSPixelDAQMappingESSourceXML::setIntervalFor(const edm::eventsetup::EventSetupRecordKey &key,
+						     const edm::IOVSyncValue& iosv, edm::ValidityInterval& oValidity)
+{
+  LogVerbatim("CTPPSPixelDAQMappingESSourceXML")
+    << ">> CTPPSPixelDAQMappingESSourceXML::setIntervalFor(" << key.name() << ")";
+
+  LogVerbatim("CTPPSPixelDAQMappingESSourceXML")
+    << "    run=" << iosv.eventID().run() << ", event=" << iosv.eventID().event();
+
+  currentBlockValid = false;
+  for (unsigned int idx = 0; idx < configuration.size(); ++idx)
+    {
+      const auto &bl = configuration[idx];
+
+      EventID startEventID = bl.validityRange.startEventID();
+      if (startEventID == EventID(1, 0, 1))
+	startEventID = EventID(1, 0, 0);
+
+      if (startEventID <= iosv.eventID() && iosv.eventID() <= bl.validityRange.endEventID())
+	{
+	  currentBlockValid = true;
+	  currentBlock = idx;
+
+	  const IOVSyncValue begin(startEventID);
+	  const IOVSyncValue end(bl.validityRange.endEventID());
+	  oValidity = ValidityInterval(begin, end);
+
+	  LogVerbatim("CTPPSPixelDAQMappingESSourceXML")
+	    << "    block found: index=" << currentBlock
+	    << ", interval=(" << startEventID << " - " << bl.validityRange.endEventID() << ")";
+
+	  return;
+	}
+    }
+
+  if (!currentBlockValid)
+    {
+      throw cms::Exception("CTPPSPixelDAQMappingESSourceXML::setIntervalFor") <<
+	"No configuration for event " << iosv.eventID();
+    }
+}
+
+
+
+CTPPSPixelDAQMappingESSourceXML::~CTPPSPixelDAQMappingESSourceXML()
+{
+}
+
+
+
+string CTPPSPixelDAQMappingESSourceXML::CompleteFileName(const string &fn)
+{
+  FileInPath fip(fn);
+  return fip.fullPath();
+}
+
+
+
+std::unique_ptr<CTPPSPixelDAQMapping> CTPPSPixelDAQMappingESSourceXML::produceCTPPSPixelDAQMapping( const CTPPSPixelDAQMappingRcd & )
+{
+  assert(currentBlockValid);
+
+  auto mapping = std::make_unique<CTPPSPixelDAQMapping>();
+  auto mask = std::make_unique<CTPPSPixelAnalysisMask>();
+
+// initialize Xerces
+  try
+    {
+      XMLPlatformUtils::Initialize();
```

Comment:
@fabferro I believe it is recommended within CMSSW to use `cms::concurrency::xercesInitialize();`
Maybe, Ianna can provide more guidance here.